packages/cli/lib/lib/sw.js

```diff
@@ -0,0 +1,54 @@
+self.__precacheManifest = [].concat(self.__precacheManifest || []);
+
+/* global workbox */
+workbox.precaching.suppressWarnings();
+/** We are sure brotli is enabled for browsers supporting script type=module
+ * so we do brotli support only for them.
+ * We can do brolti support for other browsers but there is no good way of
+ * feature detect the same at the time of pre-caching.
+ */
+if (process.env.ENABLE_BROTLI && process.env.ES_BUILD) {
+  // Alter the precache manifest to precache brotli files instead of gzip files.
+  self.__precacheManifest = self.__precacheManifest.map(asset => {
+    if (/.*.js$/.test(asset.url)) {
+      asset.url = asset.url.replace(/.esm.js$/, ".esm.js.br");
+    }
+    return asset;
+  });
+
+  class BrotliRedirectPlugin {
+    // Before saving the response in cache, we need to treat the headers.
+    async cacheWillUpdate({request, response}) {
+      if (/.js.br$/.test(request.url)) {
+        const headers = response.headers;
+        const newHeaders = {};
+        for (let key of headers.keys()) {
+          newHeaders[key] = headers.get(key);
+        }
+        // make sure the content type is compatible
+        newHeaders["content-type"] = "application/javascript";
+        return new Response((await response.text()), {
+          headers: newHeaders,
+        });
+      }
+      return response;
+    }
+  }
+  workbox.precaching.addPlugins(new BrotliRedirectPlugin());
+}
+
+const precacheOptions = {};
+if (process.env.ENABLE_BROTLI) {
```

Comment:
Should this wrap the plugin definition+invocation so that it gets stripped of out non-brotli builds?

Reply:
It kinda is already being stripped out

With `--brotli`
<img width="569" alt="screen shot 2018-12-09 at 8 34 49 pm" src="https://user-images.githubusercontent.com/5960497/49711066-1d7a1e80-fbf2-11e8-9e4d-f60951eb6b00.png">


Without --brotli
<img width="603" alt="screen shot 2018-12-09 at 8 36 09 pm" src="https://user-images.githubusercontent.com/5960497/49711071-24a12c80-fbf2-11e8-850f-cfadf97d9667.png">